### ISSUE TITLE
Not able to reproduce detection of balloons or shapes; stuck with long, thin or very small detection boxes

### ISSUE THREAD

**Issue by CarolineHaslebacher** [matterport/Mask_RCNN#2684]
I am using:

Mask-RCNN cloned from https://github.com/leekunhee/Mask_RCNN
Tensorflow 2.6.0
Keras 2.6.0
scikit-image 0.18.3
python 3.7.11
CUDA 11.1
cuDNN 8201

GPU:
NVIDIA GeForce RTX 3080

I tried to reproduce the balloon and the shapes detection by running balloon.py and train_shapes.ipynb.
If I just run the train_shapes.ipynb and do not change any parameters, so meaning I train for 1 or 2 epochs, I get this outcome:
![Screenshot](https://user-images.githubusercontent.com/56937273/131682684-cb0ea825-42ae-4053-8f85-823a0a87662f.png)
or this:
![Screenshot](https://user-images.githubusercontent.com/56937273/131685410-735db606-81b1-4940-a259-1ac5ca70bafb.png)

If I disable the mini_mask (`USE_MINI_MASK = False`), I still get an awful result:
![Screenshot](https://user-images.githubusercontent.com/56937273/131681240-ecdaef0c-5d8a-4032-a700-bde2c838bdc2.png)

If I run the ballon.py code as it came (just tweaking one line in the data loader, see #2592), for 30 epochs, I get predictions like this:
![Screenshot](https://user-images.githubusercontent.com/56937273/131682240-e764c06a-6e82-46dc-98e7-8130fff442b4.png)

Losses look good in general (blue is from train_shapes.py, orange is from balloon.py):
![Screenshot](https://user-images.githubusercontent.com/56937273/131686395-5283d397-c185-44a4-b6b0-6b53ee7e3add.png)

If I just display the data with its ground truth mask, it looks fine. To me it seems that there is something going on with the anchors and boxes.

Anyone has any idea what is going wrong? Or an idea on how to debug?

Any help is highly appreciated!

**Comment by Roharui** [comment]
https://github.com/matterport/Mask_RCNN/issues/2670#issuecomment-903080096

see here i think it's because of tensorflow version

**Comment by CarolineHaslebacher** [comment]
Thank you! Downgrading to TF version 2.4 solved these problems for me. However, I had to retrain also on version 2.4, otherwise I got no mask segmentations.

**Comment by avinash-218** [comment]
Downgrading tensorflow to 2.5.0 and h5py worked for me.

**Comment by avinash-218** [comment]
The ‘model.load_weights’ seem to load the weights incorrectly due to version compatibility issues, resulting in training from scratch. So during training and evaluation, the coco and the previously trained weights were not loaded properly and hence in case of training, the training happens from scratch and in case of evaluation the loaded model predicts worse on the sample data.

Because of this reason, the losses at earlier steps at earlier epochs were too high and also the visual results looked random, not even close to the ground truth, and also the evaluation metrics such as mAP, mAR, F1 were 0. This can be solved by two ways : 
By using ‘tf.keras.Model.load_weights’ instead of ‘model.load_weights’ - But still this can’t be used since it doesn’t support the ‘exclude’ argument.


By downgrading tensorflow from 2.7 to 2.5 worked in both training (from coco using exclude argument and from previously trained weights) and also in evaluations.

This worked for me. Correct me if i am wrong or my understanding is wrong
